virttest/cartesian_config.py

```diff
@@ -816,6 +816,15 @@ def apply_to_dict(self, d):
             d[self.name] = _substitution(self.value, d)
 
 
+class LLazySet(LOperators):
+    __slots__ = []
+    identifier = "~="
+
+    def apply_to_dict(self, d):
+        if self.name not in _reserved_keys and self.name not in d:
+            d[self.name] = _substitution(self.value, d)
```

Comment:
Should we also add regex support here? The ``LRegExpSet`` operator (``?=``) is setting a value only if present but is also using regex for the key on top of this. It might be a good idea to add support for the same here for better compatibility since your operator is more or less the reverse so one would expect similar behavior.

Reply:
Emmm, seems not possible, if the regex is mismatched it doesn't even know which keys should be set, because the given pattern cannot be used to identify them.

Reply:
Oh, I was thinking of something more like `param1(_vm.*)? ~= val1` to not set anything if `param1_vm1` exists and set `param1` if it doesn't (the regex is not matched) but I guess this will be too complicated to implement anyway.

Reply:
The problem is how could it know `param1` should be set in this case? we can't roughly assume that the target parameter to be set is coming from the given pattern by trimming wildcards or sub-patterns. Take `foo\d+_bar(_vm.*)` as an example, it's hard to guess what user wants to have because there are many possibilities.

Reply:
Yes, this is clear which is why I agreed and meant that adding extra assumptions on what the user wants and some default logic is too complicated to implement. Feel free to resolve this, I just don't have the "Resolve" button but only a "Comment" button.

Reply:
Ok, thanks!